Silicon/NXP/iMXPlatformPkg/Drivers/SmbiosConfigDxe/SmbiosConfigDxe.c

```diff
@@ -0,0 +1,712 @@
+  }
+  if ((KeyLength > MAX_VARIABLE_SIZE) || (ValueLength > MAX_VARIABLE_SIZE)) {
+    Status = EFI_INVALID_PARAMETER;
+    goto Exit;
+  }
+  
+  TempKey = AllocatePool(KeyLength + 1);
+  if (TempKey == NULL) {
+    Status = EFI_OUT_OF_RESOURCES;
+    goto Exit;
+  }
+  CopyMem(TempKey, KeyStart, KeyLength);
+  TempKey[KeyLength] = '\0';
+  KeyLength++;
+
+  TempValue = AllocatePool (ValueLength + 1);
+  if (TempValue == NULL) {
+    Status = EFI_OUT_OF_RESOURCES;
+    goto Exit;
+  }
+  CopyMem (TempValue, ValueStart, ValueLength);
+  TempValue[ValueLength] = '\0';
+  ValueLength++;
+
+  *Key = TempKey;
+  *Value = TempValue;
+  *Offset = CurrentOffset;
+
+Exit:
+  return Status;
+}
+
+#define MAX_SMBIOS_KEY_REV_1 5
+CHAR8* mSmbiosKeyRev1[MAX_SMBIOS_KEY_REV_1] = {"System Manufacturer",
+                                               "System Product Name",
+                                               "System SKU",
+                                               "System Family",
+                                               "Baseboard Product"};
+
+EFI_STATUS
+EFIAPI
+GetNextJsonKeyword (
+  CHAR8 *Buffer,
+  UINTN *Offset,
+  UINTN BufferSize,
+  CHAR8 *OutChar
+  )
+{
+  UINTN CurrentOffset;
+  EFI_STATUS Status;
+
+  Status = EFI_NOT_FOUND;
+  CurrentOffset = *Offset;
+
+  while (CurrentOffset < BufferSize) {
+    if ((Buffer[CurrentOffset] == '{') ||
+        (Buffer[CurrentOffset] == '}') ||
+        (Buffer[CurrentOffset] == '"') ||
+        (Buffer[CurrentOffset] == ',')) {
+      *OutChar = Buffer[CurrentOffset];
+      Status = EFI_SUCCESS;
+      break;
+    }
+    CurrentOffset++;
+  }
+  *Offset = CurrentOffset;
+
+  return Status;
+}
+
+EFI_STATUS
+EFIAPI
+GetSmbiosOverride (
+  CHAR8 *Key,
+  CHAR8 **Value
+  )
+{
+  SMBIOS_OVERRIDE_NODE *Node;
+  EFI_STATUS Status;
+  INTN CompareResult;
+  CHAR16 UnicodeDebugString[MAX_VARIABLE_SIZE];
+  
+  Node = mSmbiosOverrideListHead;
+  Status = EFI_NOT_FOUND;
+  while (Node != NULL) {
+    CompareResult = AsciiStrnCmp (Node->Key, Key, AsciiStrnSizeS(Key, MAX_VARIABLE_SIZE));
+    if (CompareResult == 0) {
+      AsciiStrToUnicodeStrS (Node->Key, UnicodeDebugString, MAX_VARIABLE_SIZE);
+      DEBUG ((DEBUG_INFO, "%a: Found Smbios Override Key = %s\n", __FUNCTION__, UnicodeDebugString));
+      AsciiStrToUnicodeStrS (Node->Value, UnicodeDebugString, MAX_VARIABLE_SIZE);
+      DEBUG ((DEBUG_INFO, "%a: Smbios Override Value = %s\n", __FUNCTION__, UnicodeDebugString));
+      *Value = Node->Value;
+      Status = EFI_SUCCESS;
+      goto Exit;
+    }
+    Node = Node->Next;
+  }
+
+Exit:
+  return Status;
+}
+
+EFI_STATUS
+EFIAPI
+ValidateRevision1 (
+  UINTN Count
+  )
+{
+  CHAR8 *Value;
+  EFI_STATUS Status;
+  INTN i;
+  CHAR16 UnicodeDebugString[MAX_VARIABLE_SIZE];
+
+  Status = EFI_NOT_FOUND;
+
+  // Remove Signature and Revision
+  Count -= 2;
+  if (Count != MAX_SMBIOS_KEY_REV_1) {
+    Status = EFI_NOT_FOUND;
+    goto Exit;
+  }
+
+  for (i = 0; i < MAX_SMBIOS_KEY_REV_1; i++) {
```

Comment:
Don't need to do this check. Just insert the values that we care about into the list.